internal/signature/sign.go

```diff
@@ -67,12 +75,21 @@ type Identity interface {
 	Close()
 }
 
+// SignResponse is the response from Sign containing the signature and other related metadata.
+type SignResponse struct {
+	Signature []byte
+	Cert      *x509.Certificate
+	// LogEntry is the Rekor tlog entry from the signing operation.
+	// This is only populated if offline signing mode was used.
+	LogEntry *models.LogEntryAnon
+}
+
 // Sign signs a given payload for the given identity.
```

Comment:
It also uploads to Rekor if `opts.RekorAddr` is provided!

Reply:
Done.